pkg/maps/auth/auth_map.go

```diff
@@ -0,0 +1,116 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package auth
+
+import (
+	"fmt"
+	"sync"
+	"unsafe"
+
+	"golang.org/x/sys/unix"
+
+	"github.com/cilium/cilium/pkg/bpf"
+	"github.com/cilium/cilium/pkg/ebpf"
+	"github.com/cilium/cilium/pkg/identity"
+	"github.com/cilium/cilium/pkg/policy"
+)
+
+const (
+	MapName    = "cilium_auth_map"
+	MaxEntries = 512000
+)
+
+var (
+	authMap     *Map
+	authMapInit = &sync.Once{}
+)
+
+type Map struct {
+	*ebpf.Map
+}
+
+func AuthMap() *Map {
+	authMapInit.Do(func() {
+		authMap = newAuthMap(MapName)
+	})
+	return authMap
+}
+
+func newAuthMap(mapName string) *Map {
+	return &Map{Map: ebpf.NewMap(&ebpf.MapSpec{
+		Name:       mapName,
+		Type:       ebpf.Hash,
+		KeySize:    uint32(unsafe.Sizeof(AuthKey{})),
+		ValueSize:  uint32(unsafe.Sizeof(AuthInfo{})),
+		MaxEntries: uint32(MaxEntries),
+		Flags:      unix.BPF_F_NO_PREALLOC,
+		Pinning:    ebpf.PinByName,
+	})}
+}
+
+func (m *Map) Update(localIdentity identity.NumericIdentity, remoteIdentity identity.NumericIdentity, remoteNodeID uint16, authType policy.AuthType, expiration int64) error {
+	key := newAuthKey(localIdentity, remoteIdentity, remoteNodeID, authType)
+	val := AuthInfo{Expiration: expiration}
+	return m.Map.Update(key, val, 0)
+}
+
+// AuthKey implements the bpf.MapKey interface.
+//
+// Must be in sync with struct auth_key in <bpf/lib/common.h>
+// +k8s:deepcopy-gen=true
+// +k8s:deepcopy-gen:interfaces=github.com/cilium/cilium/pkg/bpf.MapKey
+type AuthKey struct {
+	LocalIdentity  uint32 `align:"local_sec_label"`
+	RemoteIdentity uint32 `align:"remote_sec_label"`
+	RemoteNodeID   uint16 `align:"remote_node_id"`
+	AuthType       uint8  `align:"auth_type"`
+	Pad            uint8  `align:"pad"`
+}
+
+func (r *AuthKey) GetKeyPtr() unsafe.Pointer { return unsafe.Pointer(r) }
+
+func (r *AuthKey) NewValue() bpf.MapValue { return &AuthInfo{} }
+
+func (r *AuthKey) String() string {
+	return fmt.Sprintf("%d --> %d (%d)", r.LocalIdentity, r.RemoteIdentity, r.RemoteNodeID)
```

Comment:
This is mutual auth, so maybe `<->` instead of `-->` ?

Reply:
changed it to a uniform property based version -> `localIdentity=..., remoteIdentity=...`